woocommerce/woocommerce
# Add support for URLs without a port to test:e2e

This is a followup PR to #27113 

### All Submissions:


### Changes proposed in this Pull Request:

This PR updates the `test:e2e` script to support URLs without a port.

### How to test the changes in this Pull Request:

1. Set up a new test install on a host with a URL that doesn't have a port (eg. ephemeral if your local dev environment uses ports) 
2. Run the following CLI commands on that install:
```
wp plugin install woocommerce --activate 
wp theme install twentynineteen --activate
wp user create customer customer@woocommercecoree2etestsuite.com --user_pass=password --role=subscriber
wp post create --post_type=page --post_status=publish --post_title='Ready' --post_content='E2E-tests.'
```
3. In your local repo in this branch:
  - Copy `tests/e2e/env/config/default.json` to  `tests/e2e/config/default.json` 
  - Update the URL to your test install. Ensure the URL ends with a trailing `/`
  - Update the `admin` user/password to be the dev install login credentials
4. Run the following CLI commands:
```
npm install
composer install
grunt assets
npm run build:packages
```
5. Run the test suite with `npm run test:e2e`
6. **Note:** The purpose of this test is to ensure the test sequencer runs correctly. You may see some test failures (eg. on ephemeral I had REST API calls `401` so about half the tests failed.) This PR doesn't change any of the tests.

### Other information:




### Changelog entry

> Add support for URLs without a port to `npm run test:e2e`.
